### DISCUSSION TITLE
How to Use a Custom Protocol on Android?

### DISCUSSION THREAD

**Discussion by aiueo13** [tauri-apps/tauri#10378] [Q&A]
Hello, I am a beginner. And using English translator.

I want to use a custom protocol on Android. 
I'm using 'tauri::Builder::register_uri_scheme_protocol' with the 'uri_scheme' set to "test", and 'protocol' returns a PNG image.

On macOS, ＜img src="test:" /＞ works successfully.
In Windows, It looks like that ’src' need to be set to “https://test./“ insted of “test:”.

Then, What should I do on Android?
Or, Is this feature currently unimplemented?

**Comment by FabianLars** [comment]
Android behaves like Windows here so it's `http://<scheme>.`

**Reply by aiueo13** [reply]
Thank you for your response. 
However, using it causes the app to crash on Android.

You can reproduce the issue by following these steps.
1: Creating a project with 'yarn create tauri-app --beta'.
2: Adding '.register_uri_scheme_protocol("test", |_, _| Response::builder().body(Vec::new ()).unwrap())’ to 'tauri::Builder::default()' .
3: Calling 'await fetch("http://test.")' in frontend. (then crash)


```
[✔] Environment
    - OS: Mac OS 14.2.1 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.79.0 (129f3b996 2024-06-10)
    ✔ cargo: 1.79.0 (ffa9cf99a 2024-06-03)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 20.10.0
    - pnpm: 8.15.6
    - yarn: 1.22.21
    - npm: 10.2.3

[-] Packages
    - tauri [RUST]: 2.0.0-beta.24
    - tauri-build [RUST]: 2.0.0-beta.19
    - wry [RUST]: 0.41.0
    - tao [RUST]: 0.28.1
    - tauri-cli [RUST]: 2.0.0-beta.20
    - @tauri-apps/api [NPM]: 2.0.0-beta.15
    - @tauri-apps/cli [NPM]: 2.0.0-beta.22

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../build
    - devUrl: http://localhost:1420/
    - framework: Svelte
    - bundler: Vite
```

**Reply by FabianLars** [reply]
Would you mind opening a github issue with a reproduction repo for this?

**Reply by aiueo13** [reply]
Yes. I am going to bed now and will create it when I wake up!

**Reply by 0x1af2aec8f957** [reply]
I had the [same problem](https://github.com/tauri-apps/tauri/issues/10391), did this solve it?

**Reply by aiueo13** [reply]
> I had the [same problem](https://github.com/tauri-apps/tauri/issues/10391), did this solve it?

I don't know if security is safe, but I was able to do the following.
```
let uri = match TARGET_OS {
      Os::Macos => format!("{}://localhost", uri_scheme),
      Os::Android => format!("http://{}.localhost", uri_scheme),
};
```

**Reply by 0x1af2aec8f957** [reply]
This works, but now I have [a new problem](https://github.com/tauri-apps/tauri/issues/11823#issue-2703904774), can you help me?
